specification/resources/resource-manager/Microsoft.Resources/stable/2024-07-01/resources.json

```diff
@@ -1353,18 +1368,18 @@
         ],
         "responses": {
           "200": {
-            "description": "OK - Returns the validation result.",
+            "description": "The template deployment validation succeeded. Please inspect 'warnings' property since some resources might have been skipped from validation.",
             "schema": {
-              "$ref": "#/definitions/DeploymentValidateResult"
+              "$ref": "#/definitions/DeploymentExtended"
```

Comment:
Hm. I look at this line of the diff and wonder why, as in "Why DeploymentExtended? What was wrong with calling it a validation result?"

Reply:
Also am I misreading this, or are the types quite different?

Reply:
Need to restore the previous definition, as I need to return either error or success properties.

Reply:
yeah types are different, before if you notice, for 200 and 400 we returned the exact same property, now for 400 I am returning `DeploymentValidateError` and for 200 I am returning `DeploymentExtended` now is more aligned to what the service returns.